assets/src/edit-story/elements/text/output.js

```diff
@@ -39,22 +46,39 @@ function TextOutput({
     lineHeight,
     padding,
     textAlign,
+    textDecoration,
+    textOpacity,
   },
 }) {
   const style = {
     fontSize: `${dataToEditorY(fontSize, 100)}%`,
     fontStyle: fontStyle ? fontStyle : null,
     fontFamily: generateFontFamily(fontFamily, fontFallback),
     fontWeight: fontWeight ? fontWeight : null,
+    background: backgroundOpacity
+      ? rgba(backgroundColor, backgroundOpacity / 100)
+      : backgroundColor,
+    color: textOpacity ? rgba(color, textOpacity / 100) : color,
     lineHeight,
-    letterSpacing: letterSpacing ? letterSpacing + 'em' : null,
-    padding: padding ? padding + '%' : null,
+    letterSpacing: isNaN(letterSpacing) ? null : letterSpacing + 'em',
+    padding: padding
+      ? `${dataToEditorY(padding.vertical, 100)}% ${dataToEditorX(
```

Comment:
@dvoytenko Does this seem correct for converting to `%` for the output? It's currently based on how the rest of the values seem to be converted, however, it doesn't seem quite right when testing. Would you mind checking if there's something I'm missing here?

Reply:
Oops. This is actually a lot harder than just `%` for two reasons:

1. These `%` values are relative to the element itself, not the top container (layer in our case).
2. Both horizontal and vertical paddings are expressed as `%` of the _width_. 

We need to find another way to recalculate here. I think horizontal padding should be simple enough: it's `%` renormalized to the container. E.g.

```
horizontalPadding = dataToEditorX(padding.horizontal, 100) / (box.width / 100)
```

The vertical padding might be harder - we'll have to recalculate it via width. Try this first and let me know if that works:

```
const boxV = getBox({...element, height: element.width}, 100, 100);
varticalPadding = dataToEditorY(padding.vertical, 100) / (boxV.height / 100)
```

This kind of bends the mind a bit, but I think this is, at least, the right direction.

Reply:
It seems that `horizontalPadding = dataToEditorX(padding.horizontal, 100) / (box.width / 100)`
should be basically the same as directly using the width:
`dataToEditorX(padding.horizontal, width)`?

Thinking about it, shouldn't the vertical padding also be found just using `dataEditorX` and the `width`, e.g.
`dataToEditorX(padding.vertical, width)`?

If I'm testing correctly then it seems to be accurate, I might be wrong though.